tests/nn_test.py

```diff
@@ -37,7 +38,138 @@
 config.parse_flags_with_absl()
 
 
+@jtu.with_config(jax_legacy_prng_key="allow")
 class NNFunctionsTest(jtu.JaxTestCase):
+  @parameterized.parameters(False, True)
+  @jtu.skip_on_flag("jax_skip_slow_tests", True)
+  def testSdpa(self, use_bias):
+    if not jtu.test_device_matches(["gpu"]):
+      raise unittest.SkipTest("Test fails on GPU")
+
+    sdpa = nn.scaled_dot_product_attention
+    B, S, T, N, H = 4, 128, 128, 8, 32
+    keys = random.split(random.PRNGKey(0), 4)
+    Q = random.normal(keys[0], (B, T, N, H), dtype=jnp.bfloat16)
```

Comment:
We need tests for more dtypes. For example for FP32: is cudnn flashattention using TF32 by default to match the JAX implementation?

Reply:
cudnn flash attn only supports fp16 and bf16 (see [here](https://github.com/google/jax/blob/main/jax/_src/cudnn/fused_attention_stablehlo.py#L227)). So we focus on these two cases at this moment.